Makefile

```diff
@@ -31,7 +31,10 @@ DOCKER_HOST?=unix:///var/run/docker.sock
 DOCKER_VOLUME:=$(DOCKER_HOST:unix://%=%)
 
 IMAGE_REGISTRY?=gcr.io/$(PROJECT_ID)/prometheus-engine
-TAG_NAME?=$(shell date "+gmp-%Y%d%m_%H%M")
+ifeq ($(origin TAG_NAME), undefined)
+  TAG_NAME:=$(shell date "+gmp-%Y%d%m_%H%M%S_%N")
+endif
+export TAG_NAME
```

Comment:
Can we just do this?
```
BUILD_DATE:=$(shell date "+%Y%d%m_%H%M%S_%N")
export TAG_NAME?=gmp-${BUILD_DATE}
```

Reply:
I believe so. They point out it is equivalent in: https://www.gnu.org/software/make/manual/html_node/Setting.html

Reply:
I investigated this and they're not 1:1.

With yours, we will always evaluate `BUILD_TIME`. If someone uses it and calls `$(MAKE)`, the value will be different in each execution in each target.

The reason why mine works is because my if statement says if the variable is not set, then set it. In your example, `TAG_NAME` is lazily evaluated which also means you also can't simply as `export TAG_NAME?=gmp-$(shell date "+%Y%d%m_%H%M%S_%N")`, since it's re-evaluated each time you call `$(MAKE)`. While your version is simpler I feel it's more error-prone in these 2 ways.

As such, I kept my original version. Any strong thoughts here?